OffTheRecordApp/otr_shared_backend_generated_client
# Move everything to `devDependencies`

None of the packages listed in `dependencies` are actual dependencies. Those dependencies are only needed for generating the client. Moving them into `devDependencies` prevents them being added to consumers of this package. 
Most of the stuff we deploy now is bundled, so they wouldn't be added anyways. But, this will avoid issues when we need to build the package. What sparked this change is the two versions of typescript that were added when upgrading the version of typescript for one of the consumers of this package (example from the mono repo after upgrading typescript to 5.x):

```├─┬ otr-lawfirm-portal@1.0.0 -> ./apps/otr_lawfirm_web_app
│ ├─┬ @otr-app/shared-backend-generated-client@2.3.280
│ │ └── typescript@4.9.5
│ └─┬ karma-typescript@5.4.0
│   └── typescript@5.5.4 deduped invalid: "1 || 2 || 3 || 4" from node_modules/karma-typescript
```


# How to test

1. run `npm i && npm run generate:devo-angularjs`
2. Delete `node_modules` _important! on install step, in the dependent modules npm will only install `dependencies` not `devDependencies`_
3.  Run `npm link` in the this repo
4. In an app that depends on this run: `npm i @otr-app/shared-backend-generated-client`
5. Run the app and make sure everything is working as expected
